src/csharp/Microsoft.Spark/Utils/JvmObjectUtils.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Spark.Interop.Ipc;
+using System.Collections.Generic;
+
+namespace Microsoft.Spark.Utils
+{
+    /// <summary>
+    /// JvmObjectUtils is used to hold basic general helper functions that
+    /// are used within ML scope.
+    /// </summary>
+    internal class JvmObjectUtils
+    {
+        /// <summary>
+        /// Helper function for constructing the mapping between java class name and dotnet class type.
+        /// </summary>
+        /// <param name="parentType">The parent class of the target type.</param>
+        /// <param name="javaClassFieldName">The private static string field name of the dotnet class.</param>
+        /// <returns>a mapping of className and dotnet class type</returns>
+        internal static Dictionary<string, Type> ConstructJavaClassMapping(
+            Type parentType,
+            string javaClassFieldName)
+        {
+            // a mapping of java class name to the dotnet type
+            var classMapping = new Dictionary<string, Type>();
+            // search within the assemblies to find the real type that matches returnClass name
+            foreach (Assembly assembly in AppDomain.CurrentDomain.GetAssemblies())
+            {
+                foreach (Type type in assembly.GetTypes().Where(
+                    type => type.IsClass && !type.IsAbstract && type.IsSubclassOf(parentType)))
+                {
+                    FieldInfo info = type.GetField(javaClassFieldName, BindingFlags.NonPublic | BindingFlags.Static);
+                    var classNameValue = type.ContainsGenericParameters ? null : (string)info.GetValue(null);
```

Comment:
Why not move `type.ContainsGenericParameters` to the `Where` method above ?  Also update this method's description describing that generic classes are not supported.